pkg/osquery/runtime/runtime.go

```diff
@@ -831,8 +833,27 @@ func (r *Runner) launchOsqueryInstance() error {
 			case <-o.doneCtx.Done():
 				return o.doneCtx.Err()
 			case <-ticker.C:
-				if err := o.Healthy(); err != nil {
-					return errors.Wrap(err, "health check failed")
+				// Health check! Allow a couple
+				// failures before we tear everything
+				// down. This is pretty simple, it
+				// hardcodes the timing. Might be
+				// better for a Limiter
+				maxHealthChecks := 5
+				for i := 1; i <= maxHealthChecks; i++ {
+					if err := o.Healthy(); err != nil {
+						level.Info(o.logger).Log("msg", "Health check failed", "attempt", i, "err", err)
+
+						if i == maxHealthChecks {
+							return errors.Wrap(err, "health check failed")
+						}
+
+						time.Sleep(1 * time.Second)
+
+					} else {
+						// err was nil, clear failed
```

Comment:
just a thought ... should we log health check success here? Just imagining future us looking at logs and seeing a line or 2 about a failed health check ... would be nice to know if it eventually passed.

Reply:
Updated all the logging in this block. How's that look?

Reply:
Looks good other than wondering if it should be Debug or Info level. I kind a feel it should be the same as the health check above ... I don't want to see one without the other ... also I think the final attempt should before erroring out should be level.Error.